### ISSUE TITLE
Ramda propOr on object with mixed values

### ISSUE THREAD

**Issue by maxhallinan** [flow-typed/flow-typed#1176]
The type check for [`R.propOr`](https://github.com/flowtype/flow-typed/blob/master/definitions/npm/ramda_v0.x.x/flow_v0.39.x-/ramda_v0.x.x.js#L607-L611) fails when `propOr` is called on an object containing a mix of values. I'm new to Flow and static typing in general, so I'm not confident that this is a bug. However, [this test case](https://flow.org/try/#0CYUwxgNghgTiAEAzArgOzAFwJYHtXwAcYcCB5GAHgBUAaANRoEEAuAbwG0BrZgZwxiyoA5gF1mdAL4A+ABQ4YzeLXgA6NXD6LGMGFACeFAG44swKQEpmAKHjwZMgooAkAaRB6eFRlJqr1ITXhtXQNjUwt4AF4pOxwtcyiYugAfKnMbeAAyOwdnNw8vH3g4oITo+BS0gG4rUEhYBBR0bDxCYjJKWgYWDm4+AWExSVl5RWVHeFd3T29fNRUNDC0dfSMTM0tY+MSK1Jq66DgkNExcfCIScmp6JjYuXn5BUXFpOQUlXwmpgtni+MVKjUrBg9AQEAAFEAwHitSLwVgZKBCECKVDIAC2ACMoTQMqgoOiUfB+k9cRIgWA8Hx4FAIFgwETIdDYfDEcjFABGAAMHNxtnxhMUAHJGAAZACSAGEAKJCskUqkYeACokyJkw1BlGIk4RRNqXGAyIVy+BClVC8xVIA) is taken from the [`propOr` documentation](http://ramdajs.com/docs/#propOr).

```javascript
declare function propOr<T,V,A:{[k:string]:V}>(or: T, ...rest: Array<void>):
  ((p: $Keys<A>, ...rest: Array<void>) => (o: A) => V|T)
  & ((p: $Keys<A>, o: A) => V|T);
declare function propOr<T,V,A:{[k:string]:V}>(or: T, p: $Keys<A>, ...rest: Array<void>): (o: A) => V|T;
declare function propOr<T,V,A:{[k:string]:V}>(or: T, p: $Keys<A>, o: A): V|T;

type Person = {
  age: number,
  name: string,
};

const alice: Person = {
  age: 101,
  name: 'ALICE',
};

const name: (Person) => string = propOr('', 'name');
```

```
8:   age: number,
          ^ number. This type is incompatible with
9:   name: string,
           ^ string
8:   age: number,
          ^ number. This type is incompatible with
17: const name: (Person) => string = propOr('', 'name');
                            ^ string
9:   name: string,
           ^ string. This type is incompatible with
8:   age: number,
          ^ number
```

Is this a bug or an error on my part? Thanks for the help.

**Comment by maxhallinan** [comment]
Probably an error on my part. This works:

```javascript
const name: ({ name: string, }) => string = propOr('', 'name');
```

**Comment by GAntoine** [comment]
Closing since you seem to have figured it out!

**Comment by AndrewSouthpaw** [comment]
That's correct, it's a limitation of the libdef. Generally, it's *very* hard for Flow to perfectly type Ramda, because it lacks certain extra functionality.

You can take a look at the libdef for `prop`, which does allow for objects with more than one property, and see if you can extend it for `propOr`.
